lib/matplotlib/backend_bases.py

```diff
@@ -1595,6 +1595,43 @@ def __init__(self, figure):
         self.toolbar = None  # NavigationToolbar2 will set me
         self._is_idle_drawing = False
 
+    def _repr_html_(self):
+        from base64 import b64encode
+
+        fmt = self.get_default_filetype()
+        dpi = self.figure.dpi
+        if fmt == 'retina':
```

Comment:
is there really a backend that defines such a default filetype?

Reply:
`FigureCanvasBase` defaults to `rcParams['savefig.format']`. svg, pdf, pgf, ps all have this method implemented.

Reply:
Yes, but the format won't be 'retina'.

Reply:
Thank you for this. I see what I need to do. Get the ipython shell and check for retina as the figure_format. I would also need to get other settings like bbox_inches from there.

Regarding retina - it implicitly doubles the DPI, but this gives a false impression of what a saved image will look like. I think it would be better to change the rcparams. Even better would be to have a backend titled 'inline_real' and set the DPI to the DPI of the monitor (and adjust the sizes of all fonts, lines, tick lines, etc.. accordingly.).

Reply:
I am somewhat familiar with the problems of retina (well, on a theoretical PoV).  My point is just that get_default_filetype() never returns (AFAIK) "retina"; that info is simply stashed elsewhere.